sensor/kubernetes/listener/resources/secrets.go

```diff
@@ -265,78 +261,51 @@ func (s *secretDispatcher) processDockerConfigEvent(secret, oldSecret *v1.Secret
 		}
 	}
 
-	// In Kubernetes, the `default` service account always exists in each namespace (it is recreated upon deletion).
-	// The default service account always contains an API token.
-	// In OpenShift, the default service account also contains credentials for the
-	// OpenShift Container Registry, which is an internal image registry.
-	fromDefaultSA := saName == defaultSA
-	isGlobalPullSecret := secret.GetNamespace() == openshiftConfigNamespace && secret.GetName() == openshiftConfigPullSecret
+	s.processSecretForLocalScanning(secret, action, dockerConfig, saName)
 
 	newIntegrationSet := set.NewStringSet()
 	for registryAddress, dce := range dockerConfig {
 		registryAddr := strings.TrimSpace(registryAddress)
 		if registryAddr != registryAddress {
-			log.Warnf("Spaces have been trimmed from registry adress %q found in secret %s/%s",
+			log.Warnf("Spaces have been trimmed from registry address %q found in secret %s/%s",
 				registryAddress, secret.GetNamespace(), secret.GetName())
 		}
-		if fromDefaultSA {
-			// Store the registry credentials so Sensor can reach it.
-			err := s.regStore.UpsertRegistry(context.Background(), secret.GetNamespace(), registryAddr, dce)
-			if err != nil {
-				log.Errorf("Unable to upsert registry %q into store: %v", registryAddr, err)
-			}
 
-			s.regStore.AddClusterLocalRegistryHost(registryAddr)
+		registries = append(registries, &storage.ImagePullSecret_Registry{
+			Name:     registryAddr,
+			Username: dce.Username,
+		})
 
-		} else if saName == "" {
-			// only send integrations to central that do not have the k8s SA annotation
-			// this will ignore secrets associated with OCP builder, deployer, etc. service accounts
-			ii, err := DockerConfigToImageIntegration(secret, registryAddr, dce)
-			if err != nil {
-				log.Errorf("unable to create docker config for secret %s: %v", secret.GetName(), err)
-			} else if !managedcentral.IsCentralManaged() {
-				sensorEvents = append(sensorEvents, &central.SensorEvent{
-					// Only update is supported at this time.
-					Action: action,
-					Resource: &central.SensorEvent_ImageIntegration{
-						ImageIntegration: ii,
-					},
-				})
-				if features.SourcedAutogeneratedIntegrations.Enabled() {
-					newIntegrationSet.Add(ii.GetId())
-				}
-			}
+		// only send integrations to central that do not have the k8s SA annotation
+		// this will ignore secrets associated with OCP default, builder, deployer, etc. service accounts
```

Comment:
I feel like it's still not completely clear why Central would not benefit from these. These secrets are for registries only accessible from within this cluster, right? If so, then can you add that to the comment? That should clear up any confusion